src/core/app.lua

```diff
@@ -130,27 +130,63 @@ function configure (new_config)
    counter.add(configs)
 end
 
+-- Removes the claim on a name, freeing it for other programs.
+--
+-- This relinquish a claim on a name if one exists. if the name does not
+-- exist it will raise an error with an error message.
+function unclaim_name(claimed_name)
+   local name = assert(claimed_name or configuration.name, "No claim to name.")
+   local name_fq = named_program_root .. "/" .. name
+   local piddir = assert(S.readlink(name_fq))
+   local backlink = piddir .. "/name"
+
+   -- First unlink the backlink
+   assert(S.unlink(backlink))
+
+   -- Remove the actual namedir
+   assert(S.unlink(name_fq))
+
+   -- Remove from the name from the configuration
+   configuration.name = nil
```

Comment:
The reason to use a global variable is to not over-complicate things.  (It is very possible that in my ignorance I have failed to understand some essential complexity here!)  The current name already *is* effectively a global variable (the link in the file system).  A file-local "local claimed_name" would just be a cache for that.  Indeed we would only ever change it in claim_name, because it would not be exported from app.lua.  So yes I agree someone else mutating that variable would be bad -- for that reason it should be file-local and we can preserve the invariants accordingly.  (Right?)  So it's essentially just (2) which is the issue AFAIU -- could we get around this by exporting a "switch_name" function or something that would handle this?